rskj-core/src/main/java/org/ethereum/core/TransactionExecutor.java

```diff
@@ -367,9 +386,27 @@ public void go() {
 //            https://github.com/ethereum/cpp-ethereum/blob/develop/libethereum/Executive.cpp#L241
             cacheTrack.rollback();
             mEndGas = BigInteger.ZERO;
+            execError(e.getMessage());
+
+        }
+    }
+
+    public TransactionReceipt getReceipt() {
+        if (receipt == null) {
+            receipt = new TransactionReceipt();
+            long totalGasUsed = gasUsedInTheBlock + getGasUsed();
+            receipt.setCumulativeGas(totalGasUsed);
+            receipt.setTransaction(tx);
+            receipt.setLogInfoList(getVMLogs());
+            receipt.setGasUsed(getGasUsed());
+            receipt.setExecutionResult(getResult().getHReturn());
+            receipt.setError(executionError);
+//            receipt.setPostTxState(tr`ack.getRoot()); // TODO later when RepositoryTrack.getRoot() is implemented
```

Comment:
Remove this commented code,, apparently it is not needed. The receipt.setPostTxState now has a different meaning.

And this getReceipt method is only used to create ANOTHER receipt? We should review the use cases for this method.

Why not set the receipt.setPostTxState according to, maybe, executionError content?